api/CHANGELOG.md

```diff
@@ -6,6 +6,8 @@ All notable changes to this project will be documented in this file.
 
 ### :boom: Breaking Change
 
+* feat(api)!: add AnyValue and AnyValueMap types [#4575](https://github.com/open-telemetry/opentelemetry-js/pull/4575)
```

Comment:
Hmm, would you mind elaborating why this would be a breaking change (did you perhaps mean to mark it as breaking for the Logs API? :thinking: 

Since we're just adding types that don't need to be implemented by anyone it should not affect users or implementors (we've done this in the past as non-breaking when adding the Metrics API).

Reply:
I thought that adding any new export is a breaking change, but I was not 100% sure. There are three changes:

- adding the two types to the core API
- adding a new field to the experimental Event interface
- changing type of an existing field in the LogRecord interface (and Log SDK implementations)

Is only the last one a breaking change? Should I capture these as three different bullet points in the changelog?

Reply:
I have changed the API changelog to a non-breaking change.

Reply:
Adding a new export should be fine. :slightly_smiling_face: 

Thinking about it though, if we add it to the core API now, it's immediately stable, so we might want to get a few more eyes on it. 

To move ahead with this PR quicker, I'd suggest we export the `AnyValue` and `AnyValueMap` types from `@opentelemetry/api-logs` for now, as we'll integrate that it into `@opentelemetry/api` eventually when we declare it as stable. At this point it'll already be tried and tested, and it'll give us some flexibility in changing them until it's time to mark them as stable.

A side note: what's usually breaking for implementors (like this repo or dd-trace as a third party implementation) is if we add a **non-optional** field/method to an existing interface. It's not breaking for consumers of the API as they can just simply use more of the API without needing to worry that it's undefined.

- adding a new field to the experimental Event interface
  - as it's optional, this is non-breaking for both implementors and consumers
- changing type of an existing field in the LogRecord interface (and Log SDK implementations)
  - this is breaking for implementors (they will now recieve types in their implementation that they did not expect before, and type-checking will fail), but not users (they can still pass `string` to it just fine :slightly_smiling_face: )